CONTRIBUTING.md

```diff
@@ -1,47 +1,199 @@
+
+### Step 3: Fork the Repo
+
+First, create a fork of amplify-ios. Clone it, and make changes to this fork.
+
+```
+git clone git@github.com:GITHUBUSERNAME/aws-sdk-ios.git 
+# change your GITHUBUSERNAME to your github username before running this command.
+```
+
+### Step 4: Work your magic
+
+Work your magic. Here are some guidelines:
+
+- Coding style (abbreviated):
+    - In general, follow the style of the code around you
+    - 2 space indentation
+    - 120 characters wide
+    - Every change requires a new or updated unit test/integ test
+    - If you change customer facing APIs, make sure to update the documentation above the interface and include a reason for the breaking change in your PR comments
+    - Try to maintain a single feature/bugfix per pull request. It's okay to introduce a little bit of housekeeping changes along the way, but try to avoid conflating multiple features. Eventually all these are going to go into a single commit, so you can use that to frame your scope.
+
+### Step 5: Commit
+
+Create a commit with the proposed change changes:
+
+- Commit title and message (and PR title and description).
+- The title must be descriptive to the specific change.
+- No period at the end of the title.
+- Commit message should describe motivation. Think about your code reviewers and what information they need in order to understand what you did. If it's a big commit (hopefully not), try to provide some good entry points so it will be easier to follow.
+- Commit message should indicate which issues are fixed: `fixes #<issue>` or `closes #<issue>`.
+- Shout out to collaborators.
+- If not obvious (i.e. from unit tests), describe how you verified that your change works.
+- If this commit includes breaking changes, they must be listed at the top of the changelog as described above in the Pull Request Checklist.
+
+### Step 6: Pull Request
+
+- Push your changes to your GitHub fork
+- Submit a Pull Requests on the aws-sdk-ios repo to the `master` branch and add one of the amplify-ios team members on it (kneekey23, @lawmicha, @wooj2, @palpatim, @royjit, @drochetti, @phani-srikar).
+- Please follow the PR checklist written below. We trust our contributors to self-check, and this helps that process!
+- Discuss review comments and iterate until you get at least one “Approve”. When iterating, push new commits to the same branch. 
+- Usually all these are going to be squashed when you merge to master. - The commit messages should be hints for you when you finalize your merge commit message.
+- Make sure to update the PR title/description if things change. 
+- Rebase with master if the master branch has commits ahead of your fork.
+
+### Step 7: Merge
+Once your PR has been approved and tested, go ahead and merge it into `master`! It will be released in the next Amplify release which we try to do biweekly on Thursday's. Yay!! 
```

Comment:
- Will they have permission to merge to master?
- rather than saying it will be released in the next amplify release (and on a thursday), perhaps something along the lines of:

"Barring any unforeseen circumstances, your changes will be released in our next version."

I think we can avoid saying we are going to release on thurdays, etc... we are shooting for continuous releases with semantic versioning with of a release cadence of.. well.. tbd since we're still in preview.  Also, there's always exceptions w/ security patches/hot fixes, etc.

Reply:
hopefully we get to a point where we are releasing more frequently but yes will change for now